File: question/leetcode/leetcode0811.Subdomain-Visit-Count/leetcode0811.ts
function subdomainVisits(cpdomains: string[]): string[] {
	const ans: string[] = [];
	let countMap = new Map<string, number>();

	for (const cpdomain of cpdomains) {
		const [freqStr, fulldomain] = cpdomain.split(" ");
		const freq = Number(freqStr);
		const domains = fulldomain.split(".");

		for (let i = 0; i < domains.length; i++) {
			const subdomain = domains.slice(i).join(".");
			countMap.set(subdomain, (countMap.get(subdomain) || 0) + freq);
		}
	}

	for (const [domain, freq] of countMap.entries()) {
		ans.push(`${freq} ${domain}`);
	}

	return ans;
};

File: question/leetcode/leetcode0811.Subdomain-Visit-Count/leetcode0811.go
package leetcode

import (
	"fmt"
	"strconv"
	"strings"
)

func subdomainVisits(cpdomains []string) []string {
	countMap := make(map[string]int)

	for _, cpdomain := range cpdomains {
		parts := strings.Split(cpdomain, " ")
		count, _ := strconv.Atoi(parts[0])
		domain := parts[1]

		// 拆分子域名
		subdomains := strings.Split(domain, ".")
		for i := 0; i < len(subdomains); i++ {
			sub := strings.Join(subdomains[i:], ".")
			countMap[sub] += count
		}
	}

	// 构造结果
	res := []string{}
	for domain, count := range countMap {
		res = append(res, fmt.Sprintf("%d %s", count, domain))
	}

	return res
}